app/views/partials/_navbar.html.haml

```diff
@@ -1,14 +1,14 @@
 %div
   =link_to 'Home', root_path
-  = link_to 'Current Available Tasks', tasks_path
-  - if contractor_signed_in?
-    = link_to 'My Profile', edit_contractor_registration_path
-    = link_to 'Log Out', destroy_contractor_session_path, :method => :delete
-  - elsif requester_signed_in?
-    = link_to "View My Tasks", requesters_tasks_path
-    = link_to 'Log Out', destroy_requester_session_path, :method => :delete
+  = link_to 'Current Available Tasks', tasks_path( {type: 'all'})
+  - if user_signed_in? && current_user.contractor?
+    = link_to 'My Profile', edit_user_registration_path
+    = link_to 'Log Out', destroy_user_session_path, method: :delete
+  - elsif user_signed_in? && current_user.requester?
+    = link_to "View My Tasks", tasks_path({type: 'requester'})
+    = link_to 'Log Out', destroy_user_session_path, method: :delete
   - else
-    = link_to 'Requester Log In', new_requester_session_path
-    = link_to 'Requester Sign Up', new_requester_registration_path
-    = link_to 'Contractor Log In', new_contractor_session_path
-    = link_to 'Contractor Sign Up', new_contractor_registration_path
+    = link_to 'Requester Log In', new_user_session_path
+    = link_to 'Requester Sign Up', new_user_registration_path({ role: "requester"  })
```

Comment:
We tried using symbols in the user model for the requester & contractor role but RSpec tests failed so we reverted to string.